### PR TITLE
CLOUDP-150873: Cache go dependencies

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

File: .github/workflows/test-e2e.yml [hunk 1]
```diff
@@ -34,15 +41,68 @@ jobs:
...
+          docker_username: ${{ secrets.DOCKER_USERNAME }}
+          docker_password: ${{ secrets.DOCKER_PASSWORD }}
+          push_to_quay: false
+          forked: ${{ inputs.forked }}
+  e2e:
+    name: E2E tests
+    needs: [prepare-e2e, prepare-e2e-bundle]
+    runs-on: ubuntu-latest
+    env:
       DOCKER_REPO: mongodb/mongodb-atlas-kubernetes-operator-prerelease
```
Comment (Sugar-pack) [inline .github/workflows/test-e2e.yml:106]: Please add envs:  DOCKER_REGISTRY, DOCKER_REPO,  DOCKER_BUNDLES_REPO.  https://github.com/mongodb/mongodb-atlas-kubernetes/actions/runs/3793992330/jobs/6451707746#step:8:362

Comment (helderjs) [reply .github/workflows/test-e2e.yml:107]: Fixed at https://github.com/mongodb/mongodb-atlas-kubernetes/pull/802/commits/c7ddd2546faa906a50fed023e0a355a82a6890f6

**Review by Sugar-pack — APPROVED** [review]
Great work!

**Review by igor-karpukhin — APPROVED** [review]
Looks good 👍

**Pull request merged**